dd-java-agent/instrumentation/netty-4.0/src/main/java/datadog/trace/instrumentation/netty40/NettyUtils.java

```diff
@@ -0,0 +1,53 @@
+package datadog.trace.instrumentation.netty40;
+
+import datadog.trace.api.Config;
+import io.opentracing.Span;
+import io.opentracing.tag.IntTag;
+import io.opentracing.tag.Tags;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class NettyUtils {
+
+  public static final String NETTY_REWRITTEN_SERVER_STATUS_PREFIX =
+      "instrumentation.netty.server.nonstandard.http.status.";
+  public static final String NETTY_REWRITTEN_CLIENT_STATUS_PREFIX =
+      "instrumentation.netty.client.nonstandard.http.status.";
+  public static final IntTag ORIG_HTTP_STATUS = new IntTag(Tags.HTTP_STATUS.getKey() + ".orig");
+
+  private static final Map<Integer, Boolean> serverStatusCache = new HashMap<>();
```

Comment:
I'm not sure I understand the need for those caches? The `Config.getBooleanSettingFromEnvironment` will ultimately lead to a hash lookup as well (into the system properties map). Plus, since status is likely to be greater than 128, you'll be creating a new `Integer` instance for boxing every time you look up into those caches, so if this is meant for performance I'm not sure the benefit is clear vs pure simplicity.

Reply:
I was operating under the assumption that lookup by integer would have been preferable to string concatenation, lookup, and casting to boolean.  Removing as it may be adding needless complexity.